starlite/signature.py

```diff
@@ -31,9 +31,31 @@
 from starlite.utils.dependency import is_dependency_field
 from starlite.utils.typing import detect_optional_union
 
+if TYPE_CHECKING:
+    from pydantic.error_wrappers import ErrorDict
+
+
 UNDEFINED_SENTINELS = {Undefined, Signature.empty}
 
 
+def get_value_using_plugin_mapping(
```

Comment:
Perhaps this could be a `@staticmethod` on `SignatureModel` if it is not used anywhere else?

Reply:
It depends on what philosophy we want to use regarding namespaces. `signature.py` has 3 global classes and 1 function. It would be easier to parse and understand a little standalone function that is not a part of a class, but harder to get that it helps with this and only this class—so it breaks “structurality” of the code.

Let’s get back to our case. There are three viable solutions:
- keep the function the way I refactored it (with the caveat from above),
- make it a method of `PlugginMapping`,
- split `signature.py` file up into `signature_model.py` and `signature_model_factory.py`.

Reply:
What ever you think is the best one.

Reply:
I decided to go with new method in `PluginMapping`.

Reply:
Good call!